app/controllers/users_controller.rb

```diff
@@ -8,8 +8,8 @@ def index
   end
 
   def show
-    @primary_posts = @user.posts.primary.paginate(page: params[:page], per_page: 10)
-    @generated_posts = @user.posts.generated.paginate(page: params[:page], per_page: 10)
+    @primary_posts = @user.posts.primary.paginate(page: params[:page], per_page: 10).order(created_at: :desc)
```

Comment:
i think you need to put the order in front of the paginate.

`@user.posts.primary.order(created_at: :desc).paginate(page: params[:page], per_page: 10)`

Reply:
thanks @dennyluan! fixed.